runtime/tr.source/trj9/build/toolcfg/ztpf-gcc/common.mk

```diff
@@ -0,0 +1,300 @@
+# Copyright (c) 2017, 2017 IBM Corp. and others
+#
+# This program and the accompanying materials are made available under
+# the terms of the Eclipse Public License 2.0 which accompanies this
+# distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+# or the Apache License, Version 2.0 which accompanies this distribution and
+# is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# This Source Code may also be made available under the following
+# Secondary Licenses when the conditions for such availability set
+# forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+# General Public License, version 2 with the GNU Classpath
+# Exception [1] and GNU General Public License, version 2 with the
+# OpenJDK Assembly Exception [2].
+#
+# [1] https://www.gnu.org/software/classpath/license.html
+# [2] http://openjdk.java.net/legal/assembly-exception.html
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+#
+# Explicitly set shell
+#
+SHELL=/bin/sh
+
+#
+# These are the prefixes and suffixes that all GNU tools use for things
+#
+OBJSUFF=.o
+ARSUFF=.a
+SOSUFF=.so
+EXESUFF=
+LIBPREFIX=lib
+DEPSUFF=.depend.mk
+
+#
+# Paths for default programs on the platform
+# Most rules will use these default programs, but they can be overwritten individually if,
+# for example, you want to compile .spp files with a different C++ compiler than you use
+# to compile .cpp files
+#
+
+AS_PATH?=tpf-as
+M4_PATH?=m4
+SED_PATH?=sed
+AR_PATH?=ar
+PERL_PATH?=perl
+
+#
+# z/Architecture arch and tune level
+# z/TPF uses z10 as minminum supported z/Architecture.
+# mtune of z9-109 allows builtin versions of memcpy, etc.
+# instead of calling "arch-optimized" glibc functions.
```

Comment:
This rational seems backwards to me -- and setting a tune to a level older than arch is odd.  Why would tune invoke arch-optimized glibc functions instead?  Are those compatible to run on z10 (min arch)?   

I would think you are leaving some performance on the table by not tuning to z196 or newer, as that level introduced the out of order pipeline.  However, I guess this gain may not be offset by the less efficient memcpy, etc?

Reply:
OK.  Thanks for the extra insight.